main/unbound/unbound.initd

```diff
@@ -22,36 +22,21 @@ depend() {
 	after auth-dns
 }
 
-checkconfig() {
-	checkpath --directory /var/run/unbound
-	UNBOUND_PIDFILE=$("${UNBOUND_CHECKCONF}" -o pidfile "${UNBOUND_CONFFILE}")
-	return $?
-}
-
 configtest() {
-	ebegin "Checking ${SVCNAME} configuration"
-	checkconfig
-	eend $?
-}
-
-start() {
-	checkconfig || return $?
-	ebegin "Starting ${SVCNAME}"
-	start-stop-daemon --start --pidfile "${UNBOUND_PIDFILE}" \
-		--exec "${UNBOUND_BINARY}" -- -c "${UNBOUND_CONFFILE}"
+	ebegin "Checking $RC_SVCNAME configuration"
+	sed -i "s|.*pidfile:.*|pidfile:  \"$pidfile\"|" "$cfgfile"
```

Comment:
No, don’t do that. Start unbound in foreground (`-d`) and let OpenRC daemonize it and manage pidfile. That’s the job of rc system, not every program…

Reply:
it already does - see line 12

`command_args="-d $command_args"`

Reply:
`unbound` uses the pid according to it's config file so the `sed` is necessary if you want the pid set in the initd

Reply:
I added a #comment to the initd to explain the `sed`

Reply:
checksums fixed